[wasm] Bump emscripten to 3.1.12

radekdoulik opened the pull request.

radekdoulik committed:
[wasm] Bump emscripten to 3.1.9

radekdoulik committed:
Add newline

radekdoulik committed:
Use new docker images

radekdoulik committed:
Silence warnings

radekdoulik committed:
Use newer docker images

radekdoulik committed:
Bump dependencies

Commit:
Fix linking

radekdoulik committed:
Move the link fix to default link response file

To make tests build

Commit:
Disable failing test

radekdoulik committed:
Disable one more test

Affected by https://github.com/dotnet/runtime/issues/51375

radekdoulik committed:
Bump emscripten version

radekdoulik committed:
Update dependencies

radekdoulik committed:
Use newer docker images

radekdoulik commented:
/azp run runtime-wasm

radekdoulik commented:
/azp run runtime-wasm

radekdoulik commented:
/azp run runtime-wasm

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main'

radekdoulik commented:
/azp run runtime-wasm

radekdoulik commented:
@radical some tests timeout on windows, could you please take a look? like this one https://helix.dot.net/api/2019-06-17/jobs/fce081ba-bfdc-47e4-a4ae-5c8709e1cb11/workitems/normal-System.Composition.AttributeModel.Tests/console

could it be the python bug again?

radekdoulik committed:
Ignore link errors coming from native refs

This avoids this error, where the linker seems more strict now:

      [] EXEC : error : undefined symbol: _ZNKSt3__220__vector_base_commonILb1EE20__throw_length_errorEv (referenced by top-level compiled C/C++ code) [/datadisks/disk1/work/BDDD0A63/w/A3D60932/e/blz_nativeref_aot_Debug/blz_nativeref_aot_Debug.csproj]
      [] EXEC : warning : Link with `-sLLD_REPORT_UNDEFINED` to get more information on undefined symbols [/datadisks/disk1/work/BDDD0A63/w/A3D60932/e/blz_nativeref_aot_Debug/blz_nativeref_aot_Debug.csproj]
      [] EXEC : warning : To disable errors for undefined symbols use `-sERROR_ON_UNDEFINED_SYMBOLS=0` [/datadisks/disk1/work/BDDD0A63/w/A3D60932/e/blz_nativeref_aot_Debug/blz_nativeref_aot_Debug.csproj]
      [] EXEC : warning : __ZNKSt3__220__vector_base_commonILb1EE20__throw_length_errorEv may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library [/datadisks/disk1/work/BDDD0A63/w/A3D60932/e/blz_nativeref_aot_Debug/blz_nativeref_aot_Debug.csproj]
      [] EXEC : error : undefined symbol: _ZNKSt3__221__basic_string_commonILb1EE20__throw_length_errorEv (referenced by top-level compiled C/C++ code) [/datadisks/disk1/work/BDDD0A63/w/A3D60932/e/blz_nativeref_aot_Debug/blz_nativeref_aot_Debug.csproj]
      [] EXEC : warning : __ZNKSt3__221__basic_string_commonILb1EE20__throw_length_errorEv may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library [/datadisks/disk1/work/BDDD0A63/w/A3D60932/e/blz_nativeref_aot_Debug/blz_nativeref_aot_Debug.csproj]
      [] EXEC : error : Aborting compilation due to previous errors [/datadisks/disk1/work/BDDD0A63/w/A3D60932/e/blz_nativeref_aot_Debug/blz_nativeref_aot_Debug.csproj]
      [] emcc : error : '/datadisks/disk1/work/BDDD0A63/w/A3D60932/e/dotnet-workload/packs/Microsoft.NET.Runtime.Emscripten.3.1.12.Node.linux-x64/7.0.0-preview.6.22308.2/tools/bin/node

radekdoulik commented:
/azp run runtime-wasm

radical commented:
> could it be the python bug again?

No, python shouldn't be involved here. It's v8 timing out right before exit.

radekdoulik committed:
Avoid another linker error in tests

radekdoulik commented:
/azp run runtime-wasm

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main'

radekdoulik commented:
/azp run runtime-wasm

radekdoulik commented:
/azp run runtime-wasm

radekdoulik committed:
Try to use emscripten_force_exit in mono_wasm_exit

And add few debug prints

radekdoulik commented:
/azp run runtime-wasm

radekdoulik committed:
Bump docker image to get newer v8

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main'

radekdoulik committed:
Fix remaining conflict

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main'

radekdoulik committed:
Use 3.1.12 icu again, it was lost with merge

radekdoulik commented:
/azp run runtime-wasm

radekdoulik committed:
Revert "Try to use emscripten_force_exit in mono_wasm_exit"

This reverts commit bdba37b8a7ac96cead35758dc234b3d71b13880d.

radekdoulik commented:
/azp run runtime-wasm

radical commented on src/mono/wasm/wasm.proj:
```diff
@@ -137,6 +137,9 @@
       <_EmccLinkFlags Include="-Wl,--allow-undefined"/>
       <_EmccLinkFlags Include="-s ENVIRONMENT=&quot;web,webview,worker,node,shell&quot;" />
+      <!-- remove -Wno-limited-postlink-optimizations once below issue is fixed
+           active issue: https://github.com/emscripten-core/emscripten/pull/16727 -->
```
The issue seems to be fixed. And the page says "first appeared in 3.1.11"

radekdoulik replied:
It is still happening in 3.1.12, for example here https://github.com/dotnet/runtime/runs/6869929890

radical commented on src/tests/BuildWasmApps/Wasm.Build.Tests/BlazorWasmBuildPublishTests.cs:
```diff
@@ -284,6 +284,9 @@ private string CreateProjectWithNativeReference(string id)
             CreateBlazorWasmTemplateProject(id);
 
+            string extraProps = @"
+                <EmccExtraLDFlags>-sERROR_ON_UNDEFINED_SYMBOLS=0</EmccExtraLDFlags>
```
Is this still needed?

radekdoulik replied:
I think you tested it recently and it was still needed, or was it the other test?

radekdoulik replied:
Nice, we can leave just the other workaround and remove this one.

radical commented on src/native/libs/CMakeLists.txt:
```diff
@@ -144,6 +144,7 @@ if (CLR_CMAKE_TARGET_UNIX OR CLR_CMAKE_TARGET_BROWSER)
         add_compile_options(-Wno-typedef-redefinition)
         add_compile_options(-Wno-c11-extensions)
         add_compile_options(-Wno-thread-safety-analysis)
+        add_compile_options(-Wno-strict-prototypes)
```
Just curious, why was this needed?

radekdoulik replied:
The compiler is more strict and there are many new warnings. I opened https://github.com/dotnet/runtime/issues/69142 so that we can fix it in another PR.

radekdoulik committed:
Revert "Ignore link errors coming from native refs"

This reverts commit f2bfa1f22e84ecd6e9e84142ce87a97a9ecf69c3.

Let see, whether it is still needed.

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main'

radekdoulik commented:
/azp run runtime-wasm

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main'

radekdoulik commented:
/azp run runtime-wasm

radical commented:
The WBT failures seem relevant here. The ones on `main` are with nodejs, but the ones failing here are with v8.

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main' into pr-bump-emscripten-3-1-12

radekdoulik commented:
/azp run runtime-wasm

radical committed:
Merge remote-tracking branch 'origin/main' into pr-bump-emscripten-3-1-12

radical committed:
[wasm] Don't run tests on windows with v8

- We are only interested in testing with v8 on linux

This changes:
1. All the AOT tests to run with browser on windows
2. Library tests (interpreter) will run with browser, and nodejs

radical commented:
/azp run runtime-wasm

radical commented:
wasm:
library test failure is https://github.com/dotnet/runtime/issues/71843
highresource_aot failures are https://github.com/dotnet/runtime/issues/71848

radical committed:
Fix System.Net.Http.Functional.Tests failing due to trimming

.. on Windows/AOT:

```
[00:44:42] info: Discovering: System.Net.Http.Functional.Tests.dll (method display = ClassAndMethod, method display options = None)
[00:44:43] fail: System.AggregateException: One or more errors occurred. (Unable to get member, please check input for IsSupported.)
 ---> System.InvalidOperationException: Unable to get member, please check input for IsSupported.
   at Xunit.Sdk.XunitTestCase.Initialize()
   at Microsoft.DotNet.XHarness.TestRunners.Xunit.XUnitFiltersCollection.IsExcludedInternal(IEnumerable`1 , Func`2 )
   at Microsoft.DotNet.XHarness.TestRunners.Xunit.XUnitFiltersCollection.IsExcluded(ITestCase , Action`1 )
   at System.Collections.Generic.List`1[[Xunit.Abstractions.ITestCase, xunit.abstractions, Version=2.0.0.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]]..ctor(IEnumerable`1 )
   at Microsoft.DotNet.XHarness.TestRunners.Xunit.ThreadlessXunitTestRunner.Run(IEnumerable`1 )
   at Xunit.Sdk.XunitTestCase.Initialize()
   at Microsoft.DotNet.XHarness.TestRunners.Xunit.XUnitFiltersCollection.IsExcludedInternal(IEnumerable`1 , Func`2 )
   at Microsoft.DotNet.XHarness.TestRunners.Xunit.XUnitFiltersCollection.IsExcluded(ITestCase , Action`1 )
   at System.Collections.Generic.List`1[[Xunit.Abstractions.ITestCase, xunit.abstractions, Version=2.0.0.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]]..ctor(IEnumerable`1 )
   at Microsoft.DotNet.XHarness.TestRunners.Common.ApplicationEntryPoint.InternalRunAsync(LogWriter )
   at Xunit.Sdk.XunitTestCase.Initialize()
   at Microsoft.DotNet.XHarness.TestRunners.Xunit.XUnitFiltersCollection.IsExcludedInternal(IEnumerable`1 , Func`2 )
   at Microsoft.DotNet.XHarness.TestRunners.Xunit.XUnitFiltersCollection.IsExcluded(ITestCase , Action`1 )
   at System.Collections.Generic.List`1[[Xunit.Abstractions.ITestCase, xunit.abstractions, Version=2.0.0.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]]..ctor(IEnumerable`1 )
   at Microsoft.DotNet.XHarness.TestRunners.Common.ApplicationEntryPoint.InternalRunAsync(ApplicationOptions , TextWriter , TextWriter )
   at Xunit.Sdk.XunitTestCase.Initialize()
   at Microsoft.DotNet.XHarness.TestRunners.Xunit.XUnitFiltersCollection.IsExcludedInternal(IEnumerable`1 , Func`2 )
   at Microsoft.DotNet.XHarness.TestRunners.Xunit.XUnitFiltersCollection.IsExcluded(ITestCase , Action`1 )
   at System.Collections.Generic.List`1[[Xunit.Abstractions.ITestCase, xunit.abstractions, Version=2.0.0.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]]..ctor(IEnumerable`1 )
   at Microsoft.DotNet.XHarness.TestRunners.Common.WasmApplicationEntryPointBase.RunAsync()
   at Xunit.Sdk.XunitTestCase.Initialize()
   at Microsoft.DotNet.XHarness.TestRunners.Xunit.XUnitFiltersCollection.IsExcludedInternal(IEnumerable`1 , Func`2 )
   at Microsoft.DotNet.XHarness.TestRunners.Xunit.XUnitFiltersCollection.IsExcluded(ITestCase , Action`1 )
   at System.Collections.Generic.List`1[[Xunit.Abstractions.ITestCase, xunit.abstractions, Version=2.0.0.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]]..ctor(IEnumerable`1 )
   at Microsoft.DotNet.XHarness.TestRunners.Xunit.WasmApplicationEntryPoint.Run()
   at Xunit.Sdk.XunitTestCase.Initialize()
   at Microsoft.DotNet.XHarness.TestRunners.Xunit.XUnitFiltersCollection.IsExcludedInternal(IEnumerable`1 , Func`2 )
   at Microsoft.DotNet.XHarness.TestRunners.Xunit.XUnitFiltersCollection.IsExcluded(ITestCase , Action`1 )
   at System.Collections.Generic.List`1[[Xunit.Abstractions.ITestCase, xunit.abstractions, Version=2.0.0.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]]..ctor(IEnumerable`1 )
   at SimpleWasmTestRunner.Main(String[] )
   --- End of inner exception stack trace ---
```

radical commented:
/azp run runtime-wasm

radical committed:
Merge remote-tracking branch 'origin/main' into pr-bump-emscripten-3-1-12

radical commented:
/azp run runtime-wasm

radical committed:
[wasm] Fix windows/tests/aot timeout to be same for running on browser

.. as for v8.

radical committed:
Disable failing tests - issue#71877

radical commented:
/azp run runtime-wasm

radical commented:
wasm/HighResource_AOT failure is https://github.com/dotnet/runtime/issues/71848
The WBT failure is known, and unrelated to this PR.
